### PR TITLE
Add unit tests for wrong field order

### PR DESCRIPTION
**Pull request by martong**

### CODE REVIEW

File: unittests/AST/ASTImporterTest.cpp [hunk 1]
```diff
@@ -991,5 +991,80 @@ TEST_F(Fixture, TUshouldNotContainTemplatedDeclOfTypeAlias) {
...
+      getImportedDecl(
+          "struct declToImport { int a; int b; };",
+                      Lang_CXX11, "", Lang_CXX11);
+
+  MatchVerifier<Decl> Verifier;
+  ASSERT_TRUE(Verifier.match(From, cxxRecordDecl(has(fieldDecl()))));
+  ASSERT_TRUE(Verifier.match(To, cxxRecordDecl(has(fieldDecl()))));
+
+  auto Check = [](Decl *D) -> bool {
+    std::array<const char*, 2> FieldNamesInOrder = {{"a", "b"}};
```
Comment (Xazax-hun) [inline unittests/AST/ASTImporterTest.cpp]: Do you need the double braces?

File: unittests/AST/ASTImporterTest.cpp [hunk 2]
```diff
@@ -991,5 +991,80 @@ TEST_F(Fixture, TUshouldNotContainTemplatedDeclOfTypeAlias) {
...
+      };
+      )s",
+      Lang_CXX11, "", Lang_CXX11);
+
+  MatchVerifier<Decl> Verifier;
+  ASSERT_TRUE(Verifier.match(From, cxxRecordDecl(has(fieldDecl()))));
+  ASSERT_TRUE(Verifier.match(To, cxxRecordDecl(has(fieldDecl()))));
+
+  auto Check = [](Decl *D) -> bool {
+    std::array<const char*, 3> FieldNamesInOrder = {{"a", "b", "c"}};
```
Comment (Xazax-hun) [inline unittests/AST/ASTImporterTest.cpp]: Same question as above.

File: unittests/AST/ASTImporterTest.cpp (hunk 1 above)
Comment (martong) [reply unittests/AST/ASTImporterTest.cpp]: According to cppref (http://en.cppreference.com/w/cpp/container/array) we should not need the double, just the single braces. But compiling with older Clang (clang-4.0) gives the following warning:
```
../../git/llvm/tools/clang/unittests/AST/ASTImporterTest.cpp:1008:53: warning: suggest braces around initialization of subobject [-Wmissing-braces]
    std::array<const char*, 2> FieldNamesInOrder = {"a", "b"};
                                                    ^~~~~~~~
                                                    {       }
```

My solution is to skip the `=`, updated, that compiles without warning both on GCC and on Clang.

**Comment by martong** [comment]
https://reviews.llvm.org/D43967

**Pull request merged**